src/Sentry.GraphQl/SentryGraphQlHttpFailedRequestHandler.cs

```diff
@@ -0,0 +1,93 @@
+namespace Sentry.GraphQl;
+
+internal class SentryGraphQlHttpFailedRequestHandler : SentryFailedRequestHandler
+{
+    private readonly IHub _hub;
+    private readonly SentryOptions _options;
+    private readonly GraphQlContentExtractor _extractor;
+    internal const string MechanismType = "SentryGraphQLHttpFailedRequestHandler";
+    private readonly SentryHttpFailedRequestHandler _httpFailedRequestHandler;
+
+    internal SentryGraphQlHttpFailedRequestHandler(IHub hub, SentryOptions options, GraphQlContentExtractor? extractor = null)
+        : base(hub, options)
+    {
+        _hub = hub;
+        _options = options;
+        _extractor = extractor ?? new GraphQlContentExtractor(options);
+        _httpFailedRequestHandler = new SentryHttpFailedRequestHandler(hub, options);
+    }
+
+    private static readonly Regex ErrorsRegex = new ("(?i)\"errors\"\\s*:\\s*\\[", RegexOptions.Compiled);
+
+    protected internal override void DoEnsureSuccessfulResponse([NotNull]HttpRequestMessage request, [NotNull]HttpResponseMessage response)
+    {
+        JsonElement? json = null;
+        try
+        {
+            json = _extractor.ExtractResponseContentAsync(response).Result;
+            if (json is { } jsonElement)
+            {
+                if (jsonElement.TryGetProperty("errors", out var errorsElement))
```

Comment:
That gets returned by the GraphQL Server... it's part of the GraphQL protocol. You can see it in the Response.Data context in the screenshot above and the server uses it to communicate some problems with the graphql. Although it was syntactically valid GraphQL (I think the client could pick up on most errors like that before sending anything through to the server) the field `test` was not a valid field for a Notes query so the server didn't know how to serve it up.

Unlike REST, GraphQL wasn't built specifically to sit on top of HTTP so it doesn't rely on HTTP Status codes to communicate error states... it puts that stuff in the payload instead.

One question I had actually is whether we wanted to use the GraphQL Error message as the description for the event in the case of an error, since this is probably more informative than showing 'Sentry.GraphQL.SentryGraphQLHttpFailedRequestException`.

And that triggers another question, which is what we should do if there are multiple errors communicated by the GraphQL Server in the payload? Currently we capture a single exception... which I think makes sense (one request, one stack trace, one exception). What would we use as the description of the error if there were multiple exceptions though? We could generate some kind of Aggregate exception in this case (`SentryGraphQLHttpFailedRequestException` is our own fabricated exception in any case). 

And that triggers yet another question: I noticed `SentryGraphQLHttpFailedRequestException` is the only custom exception in the whole solution... which makes me nervous. Do we have a reason for avoiding our own exception types? Is there some other way preferred way to handle this?